raiden/storage/sqlite.py

```diff
@@ -292,6 +292,42 @@ def get_latest_state_change_by_data_field(
 
         return result
 
+    def get_all_state_changes(self) -> List[StateChangeRecord]:
+        """ Return all state change records (identifier and data)"""
+        cursor = self.conn.cursor()
+
+        sql = (
+            f'SELECT identifier, data '
+            f'FROM state_changes '
+            f'ORDER BY identifier '
+        )
+        cursor.execute(sql)
+
+        result = []
+        try:
+            rows = cursor.fetchall()
+            for row in rows:
+                result.append(StateChangeRecord(
+                    state_change_identifier=row[0],
+                    data=row[1],
+                ))
+        except AttributeError:
```

Comment:
What raises this exception?

Reply:
This is a good question. The whole query is a copy paste from the above function where the same exception is caught [here](https://github.com/raiden-network/raiden/blob/d05bb74d70fa196bf14bf334546014e7316fb8f2/raiden/storage/sqlite.py#L283).

Now that you mention I looked closer in both functions (and in some other functions in the file that do the same) and can't say why this is there.

Reply:
okay, so this should be removed, no?

Reply:
Hmm perhaps. As I said it's in many functions in there. @rakanalh is the other guy who pops up in the git blame. Did you add it Rakan? If yes ... why?

Reply:
I think it's because an `AttributeError` would be raised in case the `execute` fails and `fetchall` isn't a member of cursor? i couldn't find the documentation about this but i followed existing convention when it comes to why this was written this way.

Reply:
`fetchall` should always be a property of the cursor, e.g.:

```python
>>> import sqlite3
>>> conn = sqlite3.connect(':memory:')
>>> cursor = conn.cursor()
>>> cursor.fetchall
<built-in method fetchall of sqlite3.Cursor object at 0x6f897ae4da40>
>>> cursor.fetchall()
[]
```

> followed existing convention when it comes to why this was written this way.

Convention from our code base?